notification-config-service-api/src/main/proto/org/hypertrace/notification/config/service/v1/notification_config_service.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.notification.config.service.v1;
+
+import "org/hypertrace/notification/config/service/v1/notification_rule.proto";
+import "org/hypertrace/notification/config/service/v1/notification_channel.proto";
+
+service NotificationConfigService {
+  rpc CreateNotificationRule(CreateNotificationRuleRequest) returns (CreateNotificationRuleResponse) {}
+  rpc UpdateNotificationRule(UpdateNotificationRuleRequest) returns (UpdateNotificationRuleResponse) {}
+  rpc GetAllNotificationRules(GetAllNotificationRulesRequest) returns (GetAllNotificationRulesResponse) {}
+  rpc DeleteNotificationRule(DeleteNotificationRuleRequest) returns (DeleteNotificationRuleResponse) {}
+
+  rpc CreateNotificationChannel(CreateNotificationChannelRequest) returns (CreateNotificationChannelResponse) {}
```

Comment:
Should the Notification channel be a separate service - `NotificationChannelServiceImpl`? The reason being the extension of HT might have support for more channels. In that case, a different overridden service can be registered in the config service factory. Till there is a common set, it can continue to use HT one.

Reply:
yeah, makes sense to go granular.

Reply:
Separate out notification-channel in a separate module (keeping in the same pr as most of the review is done)